pkg/image/validation.go

```diff
@@ -159,5 +165,69 @@ func validateSuma(os *OperatingSystem) error {
 	if os.Suma.ActivationKey == "" {
 		return fmt.Errorf("no activation key defined")
 	}
+
+	return nil
+}
+
+func validateEmbeddedArtifactRegistry(definition *Definition) error {
+	if checkIfEmbeddedArtifactRegistryDefined(definition) {
+		return nil
+	}
+	err := validateContainerImages(definition.EmbeddedArtifactRegistry.ContainerImages)
+	if err != nil {
+		return fmt.Errorf("error validating container images: %w", err)
+	}
+	err = validateHelmCharts(definition.EmbeddedArtifactRegistry.HelmCharts)
+	if err != nil {
+		return fmt.Errorf("error validating helm charts: %w", err)
+	}
+
+	return nil
+}
+
+func checkIfEmbeddedArtifactRegistryDefined(definition *Definition) bool {
+	return len(definition.EmbeddedArtifactRegistry.HelmCharts) == 0 && len(definition.EmbeddedArtifactRegistry.ContainerImages) == 0
```

Comment:
I may totally be misreading this, but this equates to "If both of these are zero, return that yes, embedded artifact registry is defined". It feels like the logic is backwards from the name. Am I just undercaffeinated?

Reply:
nah you're right, your logic makes more sense